src/node_i18n.cc

```diff
@@ -62,6 +72,296 @@ bool flag_icu_data_dir = false;
+    SwapBytes16(dst, len * 2);
+  }
+  if (buf->IsAllocated()) {
+    MaybeLocal<Object> ret =
+        Buffer::New(isolate, reinterpret_cast<char*>(buf->out()), len);
+    buf->Release();
+    return ret;
+  }
+  return Buffer::Copy(isolate, reinterpret_cast<char*>(buf->out()), len);
+}
+
+struct Converter {
+  Converter(Environment* env, const char* name)
+      : conv(nullptr) {
+    UErrorCode status = U_ZERO_ERROR;
+    conv = ucnv_open(name, &status);
+    if (U_FAILURE(status)) {
+      ThrowICUError(env, status, "Unable to create converter");
+    }
+  }
+
+  ~Converter() {
+    if (conv != nullptr)
+      ucnv_close(conv);
+  }
+
+  UConverter* conv;
+};
+
+// One-Shot Converters
+
+// Converts the Buffer from one named encoding to another.
+// args[0] is a string identifying the encoding we're converting to.
+// args[1] is a string identifying the encoding we're converting from.
+// args[2] must be a buffer instance
+void Convert(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+
+  CHECK(args[0]->IsString());
+  CHECK(args[1]->IsString());
+  THROW_AND_RETURN_UNLESS_BUFFER(env, args[2]);
+
+  Utf8Value to_name(env->isolate(), args[0]);
+  Utf8Value from_name(env->isolate(), args[1]);
+
+  SPREAD_ARG(args[2], ts_obj);
+
+  UErrorCode status = U_ZERO_ERROR;
+
+  MaybeStackBuffer<char> buf;
+  uint32_t len, limit;
+  char* target;
+  const char* data = ts_obj_data;
+
+  Converter to(env, *to_name);
+  Converter from(env, *from_name);
+  if (to.conv == nullptr || from.conv == nullptr)
+    return;
+
+  ucnv_setSubstChars(to.conv, "?", 1, &status);
+  status = U_ZERO_ERROR;
+
+  limit = ts_obj_length * ucnv_getMaxCharSize(to.conv);
+  buf.AllocateSufficientStorage(ts_obj_length * ucnv_getMaxCharSize(to.conv));
+  target = *buf;
+
+  ucnv_convertEx(to.conv, from.conv,
+                 &target, target + limit,
+                 &data, ts_obj_data + ts_obj_length,
+                 NULL, NULL, NULL, NULL,
+                 true, true,
+                 &status);
+
+  if (U_SUCCESS(status)) {
+    len = target - *buf;
+    args.GetReturnValue().Set(
+      AsBuffer(env->isolate(), &buf, len).ToLocalChecked());
+  } else {
+    ThrowICUError(env, status, "Unable to transcode buffer");
+  }
+}
+
+// Converts to UCS2 from ISO-8859-1 and US-ASCII
+// args[0] is the encoding to convert from
+// args[1] must be a buffer instance
+void ConvertToUcs2(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+
+  CHECK(args[0]->IsString());
+  THROW_AND_RETURN_UNLESS_BUFFER(env, args[1]);
+
+  Utf8Value name(env->isolate(), args[0]);
+
+  SPREAD_ARG(args[1], ts_obj);
+
+  UErrorCode status = U_ZERO_ERROR;
+  size_t len;
+
+  MaybeStackBuffer<UChar> buf(ts_obj_length);
+
+  Converter from(env, *name);
+  if (from.conv == nullptr)
+    return;
+
+  len = ucnv_toUChars(from.conv,
+                      *buf, ts_obj_length << 1,
```

Comment:
Can you write that as `ts_obj_length * sizeof(*buf)`?  It's both more correct and more obvious.